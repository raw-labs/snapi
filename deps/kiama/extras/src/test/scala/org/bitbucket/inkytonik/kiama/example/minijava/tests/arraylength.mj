class ArrayLength {
    public static void main () {
        System.out.println (new ArrayLengthClass ().run ());
    }
}

class ArrayLengthClass {

    public int run () {
        int[] arr;
        arr = new int[10];
        return arr.length;
    }

}
