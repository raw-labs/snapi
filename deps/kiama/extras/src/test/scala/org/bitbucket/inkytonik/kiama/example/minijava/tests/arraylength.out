.source src/test/scala/org/bitbucket/inkytonik/kiama/example/minijava/tests/arraylength.mj
.class public ArrayLength
.super java/lang/Object

.method <init>()V
.limit stack 1
.limit locals 1
    aload_0
    invokespecial java/lang/Object/<init>()V
    return
.end method

.method public static main([Ljava/lang/String;)V
.limit stack 3
.limit locals 2
    getstatic java/lang/System/out Ljava/io/PrintStream;
    new ArrayLengthClass
    dup
    invokespecial ArrayLengthClass/<init>()V
    invokevirtual ArrayLengthClass/run()I
    invokevirtual java/io/PrintStream/println(I)V
    return
.end method

.source src/test/scala/org/bitbucket/inkytonik/kiama/example/minijava/tests/arraylength.mj
.class public ArrayLengthClass
.super java/lang/Object

.method <init>()V
.limit stack 1
.limit locals 1
    aload_0
    invokespecial java/lang/Object/<init>()V
    return
.end method

.method public run()I
.limit stack 2
.limit locals 2
    bipush 10
    newarray int
    astore 1
    aload 1
    arraylength
    ireturn
.end method

